### PR TITLE
Refs #26905 -- Added possibility to use any MultiValueDict-like objects as a form data

### PR DESCRIPTION
**Pull request by marcinn**
Here is a patch proposal for https://code.djangoproject.com/ticket/26905

### CODE REVIEW

**Comment by charettes** [comment]
Buildbot, add to whitelist.

File: tests/forms_tests/tests/test_forms.py [hunk 1]
```diff
@@ -855,6 +855,10 @@ class SongForm(Form):
                 widget=CheckboxSelectMultiple,
             )
 
+        class MultiValueDictLike(dict):
```
Comment (timgraham) [inline tests/forms_tests/tests/test_forms.py]: I think the new tests should be in new test methods.

File: tests/forms_tests/tests/test_forms.py [hunk 2]
```diff
@@ -867,7 +871,17 @@ class SongForm(Form):
         f = SongForm(data)
         self.assertEqual(f.errors, {})
 
+        # checking whether MultiValueDictLike.getlist() is called,
+        # even if composers value is a string
+        data = MultiValueDictLike(dict(name='Yesterday', composers='J'))
```
Comment (timgraham) [inline tests/forms_tests/tests/test_forms.py]: I'm not sure why the existing test uses `dict(...)` -- let's use `{'k': 'v'}` for the new tests.

**Comment by timgraham** [comment]
merged in 74bb013cc1edb0f2fcf24c750d4d942cae507765, thanks!

**Pull request closed without merging**